Restore and update Payment Services dropin docs (#852)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/payment-services/initialization.mdx

import TableWrapper from '@components/TableWrapper.astro';
import { Aside } from '@astrojs/starlight/components';

> The Payment Services initializer provides options for setting the backend endpoint and configuring language definitions.

Reviewer: ```suggestion
The Payment Services initializer provides configuration options for the backend endpoint and language definitions.
```

<strong>Version: 3.1.1</strong>
</div>

## Configuration options

The following table describes the configuration options available for the **Payment Services** initializer:

<TableWrapper nowrap={[0, 1, 2]}>
| Option             | Type     | Req? | Description                                                                                                                                                                                                                                                                                                                                                                                                                          |
|--------------------|----------|------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `apiUrl`           | string   | Yes  | The Adobe Commerce GraphQL endpoint URL, such as "https://example.com/graphql".                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| `apiUrl` | string | Yes | Adobe Commerce GraphQL endpoint URL (for example, `https://example.com/graphql`). |
```

> | `getCustomerToken` | function | Yes  | The Payment Services drop-in can send GraphQL requests on behalf of the shopper. This requires GraphQL authorization, which can be performed using authorization tokens or session cookies. For token-based authorization, the `getCustomerToken` function should return a customer token as a string, or null for guest customers. For session-based authorization, the `getCustomerToken` property must explicitly be set to null. |

Reviewer: ```suggestion
| `getCustomerToken` | function | Yes | Provides authorization for GraphQL requests made on behalf of the shopper. For token-based auth, it must return a customer token (string) or `null` for guests. For session-based auth, it must be set to `null`. |
```

> | `storeViewCode`    | string   | No   | The Adobe Commerce store view code against which the Payment Services drop-in should send GraphQL requests. When not set, no 'Store' HTTP header will be sent in the GraphQL requests.                                                                                                                                                                                                                                               |

Reviewer: ```suggestion
| `storeViewCode` | string | No | Adobe Commerce store view code used for GraphQL requests. If not set, the `Store` HTTP header is not included. |
```

> | `langDefinitions`  | object   | No   | Provides language definitions for internationalization (i18n).                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| `langDefinitions` | object | No | Language definitions used for internationalization (i18n). |
```

## Customizing data models

No customizable models are available for this drop-in.

## Drop-in configuration

> The **Payment Services initializer** provides options for setting the backend endpoint and configuring language definitions.

Reviewer: ```suggestion
The **Payment Services initializer** provides options to configure the backend endpoint and language definitions.
```

import { fetchPlaceholders } from '../commerce.js';
import { getConfigValue } from '@dropins/tools/lib/aem/configs.js';
import { getCookie } from '@dropins/tools/lib.js';
import { initializers } from '@dropins/tools/initializer.js';

const initializeDropin = async () => {
  const labels = await fetchPlaceholders('placeholders/payment-services.json');
  const langDefinitions = {
    default: {
      ...labels,
    },
  };

  const coreEndpoint = await getConfigValue('commerce-core-endpoint');
  const getUserTokenCookie = () => getCookie('auth_dropin_user_token');

  return initializers.mountImmediately(initialize, {
    apiUrl: coreEndpoint,
    getCustomerToken: getUserTokenCookie,
    langDefinitions,
  });
};

await initializeDropin();
```

<Aside type="note">
    Refer to the [Configuration options](#configuration-options) table for detailed descriptions of each option.
</Aside>

### Initialization is asynchronous

The Payment Services drop-in initializer is fully asynchronous. Even if mounted immediately, it initializes in the background and emits initialization updates via custom events.

### Payment method availability

> During asynchronous initialization, the Payment Services drop-in emits a series of events to signal drop-in readiness in different locations. For each of these events, the event payload is an `{ availablePaymentMethods: string[] }` object containing the available payment method codes.

Reviewer: ```suggestion
During asynchronous initialization, the Payment Services drop-in emits a series of events to indicate readiness across different locations. Each event includes a payload of `{ availablePaymentMethods: string[] }`, listing the available payment method codes.
```

The following table contains the available event names and their descriptions:

<TableWrapper nowrap={[0]}>
| Event name                                    | Description                                                                                                                                                |
|-----------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `payment-services/initialized/checkout`       | Drop-in initialized for the `CHECKOUT` location, with `event.availablePaymentMethods` determining the payment methods available for the checkout page.     |

Reviewer: ```suggestion
| `payment-services/initialized/checkout` | Indicates the drop-in is initialized for the `CHECKOUT` location. The `event.availablePaymentMethods` property lists the payment methods available on the checkout page. |
```

> | `payment-services/initialized/product-detail` | Drop-in initialized for the `PRODUCT_DETAIL` location, with `event.availablePaymentMethods` determining the payment methods available for the product details page. |

Reviewer: ```suggestion
| `payment-services/initialized/product-detail` | Indicates the drop-in is initialized for the `PRODUCT_DETAIL` location. The `event.availablePaymentMethods` property lists the payment methods available on the product detail page. |
```

</TableWrapper>

> The following code shows an example implementation of payment method availability event handlers for the checkout page.

Reviewer: ```suggestion
The following example demonstrates how to implement payment method availability event handlers for the checkout page.
```
